rest/api_test.go

```diff
@@ -3671,6 +3672,216 @@ func TestUnsupportedConfig(t *testing.T) {
 	sc.Close()
 }
 
+func TestConflictWithInvalidAttachment(t *testing.T) {
+	var rt RestTester
+	defer rt.Close()
+
+	//Create Doc
+	docrevId := rt.createDoc(t, "doc1")
+
+	docRevDigest := strings.Split(docrevId, "-")[1]
+
+	//Setup Attachment
+	attachmentBody := base64.StdEncoding.EncodeToString(make([]byte, 20))
+	attachmentContentType := "content/type"
+	reqHeaders := map[string]string{
+		"Content-Type": attachmentContentType,
+	}
+
+	//Set attachment
+	response := rt.SendRequestWithHeaders("PUT", "/db/doc1/attach1?rev="+docrevId, attachmentBody, reqHeaders)
+	var body db.Body
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docrevId2 := body["rev"].(string)
+
+	//Get Existing Doc & Update rev
+	response = rt.SendRequest("GET", "/db/doc1", "")
+	json.Unmarshal(response.Body.Bytes(), &body)
+	body["_rev"] = docrevId2
+	body["rev"] = docrevId2
+
+	//Update Doc
+	temp, err := json.Marshal(body)
+	assert.NoError(t, err)
+	str := string(temp)
+	response = rt.SendRequest("PUT", "/db/doc1", str)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docrevId3 := body["rev"].(string)
```

Comment:
Reusing the previously returned body (without indicating what that is) in this test makes it unclear what each update operation is actually doing, and would make it trickier to debug the test.  I'd recommend switching to something like TestAttachments, where the body is explicitly shown for each step.  It would also allow you to skip the marshal step in most cases, and just use something like:
```
`{"foo":"bar"}`
```